### PR TITLE
Updates in event selection criteria for Pb-Pb analysis

### PR DESCRIPTION
**Pull request by sustripathy**

### CODE REVIEW

**Review by victor-gonzalez — DISMISSED** [review]

File: PWGCF/TableProducer/filterCorrelations.cxx [hunk 1]
```diff
@@ -59,7 +59,7 @@ struct FilterCF {
   O2_DEFINE_CONFIGURABLE(cfgCutMCPt, float, 0.5f, "Minimal pT for particles")
   O2_DEFINE_CONFIGURABLE(cfgCutMCEta, float, 0.8f, "Eta range for particles")
   O2_DEFINE_CONFIGURABLE(cfgVerbosity, int, 1, "Verbosity level (0 = major, 1 = per collision)")
-  O2_DEFINE_CONFIGURABLE(cfgTrigger, int, 7, "Trigger choice: (0 = none, 7 = sel7, 8 = sel8)")
+  O2_DEFINE_CONFIGURABLE(cfgTrigger, int, 8, "Trigger choice: (0 = none, 7 = sel7, 8 = sel8, 9 = sel8 + kNoSameBunchPileup + kIsGoodZvtxFT0vsPV, 10 = sel8 before April, 2024, 11 = sel8 for MC)")
```
Comment (jgrosseo) [inline PWGCF/TableProducer/filterCorrelations.cxx:62]:
```suggestion
  O2_DEFINE_CONFIGURABLE(cfgTrigger, int, 7, "Trigger choice: (0 = none, 7 = sel7, 8 = sel8, 9 = sel8 + kNoSameBunchPileup + kIsGoodZvtxFT0vsPV, 10 = sel8 before April, 2024, 11 = sel8 for MC)")
```

File: PWGCF/TableProducer/filterCorrelations.cxx [hunk 2]
```diff
@@ -98,6 +98,12 @@ struct FilterCF {
       return collision.alias_bit(kINT7) && collision.sel7();
     } else if (cfgTrigger == 8) {
       return collision.sel8();
+    } else if (cfgTrigger == 9) { // relevant only for Pb-Pb
+      return collision.sel8() && collision.selection_bit(aod::evsel::kNoSameBunchPileup) && collision.selection_bit(aod::evsel::kIsGoodZvtxFT0vsPV);
+    } else if (cfgTrigger == 10) { // TVX trigger only (sel8 selection before April, 2024)
+      return collision.selection_bit(aod::evsel::kIsTriggerTVX);
```
Comment (jgrosseo) [inline PWGCF/TableProducer/filterCorrelations.cxx:104]: isn't this missing `collision.sel8() &&`?

File: PWGCF/Tasks/correlations.cxx [hunk 3]
```diff
@@ -107,7 +107,7 @@ struct CorrelationTask {
 
   // Track filters
   Filter trackFilter = (nabs(aod::track::eta) < cfgCutEta) && (aod::track::pt > cfgCutPt) && ((requireGlobalTrackInFilter()) || (aod::track::isGlobalTrackSDD == (uint8_t) true));
-  Filter cfTrackFilter = (nabs(aod::cftrack::eta) < cfgCutEta) && (aod::cftrack::pt > cfgCutPt);
+  Filter cfTrackFilter = (nabs(aod::cftrack::eta) < cfgCutEta) && (aod::cftrack::pt > cfgCutPt) && (aod::track::trackType & (uint8_t)cfgTrackBitMask) == (uint8_t)cfgTrackBitMask;
```
Comment (jgrosseo) [inline PWGCF/Tasks/correlations.cxx:110]:
```suggestion
  Filter cfTrackFilter = (nabs(aod::cftrack::eta) < cfgCutEta) && (aod::cftrack::pt > cfgCutPt) && ((aod::track::trackType & (uint8_t)cfgTrackBitMask) == (uint8_t)cfgTrackBitMask);
```

**Review by jgrosseo — COMMENTED** [review]
Few suggestions inline

File: PWGCF/TableProducer/filterCorrelations.cxx (hunk 2 above)
Comment (sustripathy) [reply PWGCF/TableProducer/filterCorrelations.cxx:104]: No, I think we should not have `collision.sel8()` for this option as in the new version we have additional kNoTimeFrameBorder and kNoITSROFrameBorder cuts which were not present before. 
So, we need only the first part of 
`bool sel8 = bc.selection_bit(kIsTriggerTVX) && bc.selection_bit(kNoTimeFrameBorder) && bc.selection_bit(kNoITSROFrameBorder);`

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**